### ISSUE TITLE
Redirects to /favicon.ico

### ISSUE THREAD

**Issue by xombiemp** [antifuchs/htpasswd-login#1]
Hi, I'm trying to get your example working.  It seems to be authenticating correctly, but it is redirecting to /favicon.ico after a successful login. My original request is just to the root of my domain /.

**Comment by antifuchs** [comment]
Thanks for reporting this - I'm sorry it took me ages to get to this Issue (my github notification prefs were messed up & I ended up not being subscribed to my own repo's issues)!

This is a known bug that I'm just now trying to do something effective about - this is caused by the way browsers make HTTP requests, and the way this tool communicates the "when correctly authenticated, redirect to this place" logic. I'd originally written this to support only static files, so couldn't very effectively pass the redirect URL through via the form; instead, I chose to set a session cookie with the redirection URL. Now, if a browser made a request to a URL that required authentication, it would most likely _also_ attempt to retrieve `/favicon.ico` simultaneously, which ended up being the redirect URL.

Previously I had thought that looking at "more specific" path components would fix this (999bb97a796904b92d9cd4925dee8e9469cb95da), but that's not the case.

The only way to truly fix this is to put the redirect URL into the login form - this will ensure that the user gets redirected to the URL that they requested (since the login form they are shown upon authentication failure will contain that URL as a form field).

PR #2 should fix this. I'll try and get it merged ~this weekend.
